### PR TITLE
Update depricated PF multiselect component 

### PR DESCRIPTION
**Pull request by ashley-o0o**
Closes: [RHOAIENG-421](https://issues.redhat.com/browse/RHOAIENG-421)

## Description
This PR updates the deprecated Patternfly 4 select component with the PF5 [multi select component](https://www.patternfly.org/components/menus/select/#multiple-typeahead-with-chips)

## How Has This Been Tested? 
This is tested with `npm run test` as well as on localhost, and using the Cypress tests in `userManagement.cy.ts`

## Test Impact
No new tests were added as there are no functionality changes 

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by christianvogt** [comment]
- When I first click into the input field, my focus is lost and I need to click again before I can type.
- After using the keyboard to highlight an item in the dropdown, pressing `enter` no longer selects / deselects that item.
- After focusing the text input with the keyboard, pressing an arrow key no longer opens the menu. This behavior works in the PF example.
- Previously I could not focus the arrow dropdown button. This made sense since the menu is controlled by focusing on the input field. Now I can also tab to the toggle button. The toggle button should not be focusable as per the PF example.

File: frontend/src/components/MultiSelection.tsx [hunk 1]
```diff
@@ -9,46 +23,174 @@ type MultiSelectionProps = {
...
+        isOpen={isOpen}
+        selected={selected}
+        onSelect={(ev, selection) => onSelect(value.find((option) => option.name === selection))}
+        onOpenChange={() => setIsOpen(false)}
+        toggle={toggle}
       >
-        {value.map((option, index) => (
-          <SelectOption isDisabled={false} key={index} value={option.name} />
-        ))}
+        <SelectList isAriaMultiselectable id="select-multi-typeahead-listbox">
```
Comment (jeff-phillips-18) [inline frontend/src/components/MultiSelection.tsx:177]: the id should be unique across multiple MultiSelection components

File: frontend/src/components/MultiSelection.tsx [hunk 2]
```diff
@@ -9,46 +23,174 @@ type MultiSelectionProps = {
...
-        }}
-        onClear={clearSelection}
-        selections={value.filter((element) => element.enabled).map((element) => element.name)}
-        isOpen={showMenu}
-        aria-label="Select groups menu"
-        typeAheadAriaLabel={ariaLabel}
-        isCreatable={false}
-        onCreateOption={undefined}
-        validated={noSelectedItems ? 'error' : 'default'}
+        id="multi-typeahead-select"
```
Comment (jeff-phillips-18) [inline frontend/src/components/MultiSelection.tsx:170]: the id should be unique across multiple MultiSelection components

**Review by jeff-phillips-18 — CHANGES_REQUESTED** [review]

**Comment by jeff-phillips-18** [comment]
> * Previously I could not focus the arrow dropdown button. This made sense since the menu is controlled by focusing on the input field. Now I can also tab to the toggle button. The toggle button should not be focusable as per the PF example.

This is a PF issue. Though it works on the documentation page, if you try the code sandbox it does not and the toggle button is in the tab order for focus.

File: frontend/src/components/MultiSelection.tsx [hunk 3]
```diff
@@ -9,46 +23,174 @@ type MultiSelectionProps = {
...
+      isExpanded={isOpen}
+      isFullWidth
+    >
+      <TextInputGroup isPlain>
+        <TextInputGroupMain
+          value={inputValue}
+          onClick={onToggleClick}
+          onChange={onTextInputChange}
+          onKeyDown={onInputKeyDown}
+          id="multi-typeahead-select-input"
```
[the remark this answers is not in the document]
Comment (christianvogt) [reply frontend/src/components/MultiSelection.tsx:126]: Unless referenced by some aria tag we can likely just remove all `id` attributes.

**Review by jeff-phillips-18 — APPROVED** [review]

File: frontend/src/components/MultiSelection.tsx [hunk 4]
```diff
@@ -9,46 +23,174 @@ type MultiSelectionProps = {
...
+    if (inputValue) {
+      setIsOpen(true);
+    }
+    setFocusedItemIndex(null);
+    setActiveItem(null);
+  }, [inputValue]);
+
+  const handleMenuArrowKeys = (key: string) => {
+    let indexToFocus;
+    if (isOpen) {
```
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply frontend/src/components/MultiSelection.tsx:49]: This is not resovled

File: frontend/src/components/MultiSelection.tsx [hunk 5]
```diff
@@ -9,46 +23,179 @@ type MultiSelectionProps = {
...
```
Comment (jeff-phillips-18) [inline frontend/src/components/MultiSelection.tsx:49]: Since this is back again: if it is not open, it should be opened and just return.

**Review by jeff-phillips-18 — APPROVED** [review]
/lgtm

**Review by jeff-phillips-18 — APPROVED** [review]
/lgtm

**Pull request merged**